lib/v2/ndss-symposium/ndss.js

```diff
@@ -0,0 +1,55 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const url = 'https://www.ndss-symposium.org';
+const { parseDate } = require('@/utils/parse-date');
+// 2 sytles:
+// * https://www.ndss-symposium.org/ndss2022/accepted-papers/
+// * https://www.ndss-symposium.org/ndss2021/accepted-papers/
+
+module.exports = async (ctx) => {
+    const last = new Date().getFullYear() + 1;
+    const yearList = Array.from({ length: last - 2020 }, (_, v) => `${url}/ndss${v + 2020}/accepted-papers`);
+    const reponses = await got.all(yearList.map((url) => got(url)));
+
+    const items = reponses.map((response) => {
```

Comment:
This will only match 2020 and 2021 even though 2022's site looks similar to 2020's one

Reply:
why? the `yearList` here is `['/ndss2020/accepted-papers', '/ndss2021/accepted-papers', '/ndss2022/accepted-papers']`

Reply:
I mean the selectors inside the map. Neither `$('.page-template-default section.new-wrapper p')` nor `$('div.single-paper')` can match a thing on https://www.ndss-symposium.org/ndss2022/accepted-papers/

Reply:
OK. I see...That's a little tricky. Any suggestions?

Reply:
Not really. 
For 2022 and 2020, every `<h3>` is the title and the next sibling `<p>` is the author. 
For 2021 and 2019, everything inside `<strong>` is the title and the sibling text is the author.
You can use `filter()` with `.nodeType === 3` to extract the text without any tags. (Ref: [SO](https://stackoverflow.com/a/14755309), [RSSHub](https://github.com/DIYgod/RSSHub/search?q=nodeType))

---

Apart from that, since this is a WordPress-based site, you can use the REST API that comes from WordPress and obtain multiple years of data in only [1 HTTP request](https://www.ndss-symposium.org/wp-json/wp/v2/pages?slug=accepted-papers) in JSON.

---
```js
    const { data } = await got('https://www.ndss-symposium.org/wp-json/wp/v2/pages?slug=accepted-papers');

    const items = data.map((elem) => {
        const $ = cheerio.load(elem.content.rendered);
        const link = elem.link;
        const pubDate = parseDate(elem.date);

        const divMatch = $('div h3')
            .toArray()
            .map((item) => {
                item = $(item);
                const title = item.text().trim();
                return {
                    title,
                    author: item.siblings().text().trim().replaceAll('\n', '').replace(/\s+/g, ' '),
                    link: `${link}#${title}`,
                    pubDate,
                };
            });
        const pMatch = $('p strong')
            .toArray()
            .map((item) => {
                item = $(item);
                const title = item.text().trim();
                return {
                    title,
                    author: item
                        .parent()
                        .contents()
                        .filter((_, e) => e.nodeType === 3)
                        .text()
                        .trim()
                        .replaceAll('\n', '')
                        .replace(/\s+/g, ' '),
                    link: `${link}#${title}`,
                    pubDate,
                };
            });
        return divMatch.length ? divMatch : pMatch;
    });
```